pilot/pkg/model/context.go

```diff
@@ -332,6 +332,14 @@ type Proxy struct {
 	// LastPushContext; the XDS cache depends on knowing the time of the PushContext to determine if a
 	// key is stale or not.
 	LastPushTime time.Time
+
+	// LastPushedResources contains the list of resources that were last time to the proxy,
+	// keyed by the DiscoveryRequest TypeUrl. Currently it is used only for CDS
+	LastPushedResources map[string][]*discovery.Resource
+
+	// ResourceVersionWaitingForAck holds the latest versions sent to the proxy for which ack is not received yet
```

Comment:
there can be multiple un-acked versions

Reply:
yeah, so we are keeping only the last sent and rds will be pushed only after we get ack which is for this stored pushVersion or any other push which is latest than the stored one.